src/coreclr/jit/morph.cpp

```diff
@@ -8621,8 +8624,13 @@ void Compiler::fgMorphTailCallViaJitHelper(GenTreeCall* call)
     assert(ppArg != nullptr);
     assert(*ppArg == nullptr);
 
+#ifndef TARGET_WASM
     unsigned nOldStkArgsWords =
         (compArgSize - (codeGen->intRegState.rsCalleeRegArgCount * REGSIZE_BYTES)) / REGSIZE_BYTES;
+#else
+    unsigned nOldStkArgsWords = 0;
+    assert(false); // TODO: Wasm: what to do here?
```

Comment:
This whole method is x86-specific. It should be fine to put the whole method into x86 ifdef.

Reply:
There's one problem: https://github.com/dotnet/runtimelab/blob/9b5c306f8983f90036a0f2a49c7c0d2393260c36/src/coreclr/jit/morph.cpp#L7749  What should I do with that reference?

Reply:
>  What should I do with that reference?

This path is unreachable on anything but `x86` (see `fgCanTailCallViaJitHelper`), and is the place that calls `fgMorphTailCallViaJitHelper`. I'd think `fgMorphTailCallViaJitHelper` can be left as-is (and the compiler should eliminate it anyway)?

Reply:
Right, `fgMorphTailCallViaJitHelper` is reachable on x86 only, and the compiler should be eliminating it today. It just not obvious at all from cursory look at the code. If the whole method was under ifdef, it would be much more obvious.

Reply:
Makes sense. The complication is of course that `fgCanTailCallViaJitHelper` can return `false` for methods that use `stackalloc`, which means the `else` path is used on `x86` too.

Perhaps something like the following would work (optionally `ifdef`ing out both `fgCanTailCallViaJitHelper` and `fgMorphTailCallViaJitHelper` too):
```C++
#ifdef TARGET_X86
if (fgCanTailCallViaJitHelper())
{
    tailCallViaJitHelper = true;
}
#endif

if (!tailCallViaJitHelper)
{
    // The former "else" block
}
```

Reply:
have followed the advice here.